pkg/openshift/rosa/cluster.go

```diff
@@ -0,0 +1,438 @@
+
+	return clusterID, err
+}
+
+// findCluster gets the cluster the body
+func (r *Provider) findCluster(ctx context.Context, clusterName string) (*clustersmgmtv1.Cluster, error) {
+	query := fmt.Sprintf("product.id = 'rosa' AND name = '%s'", clusterName)
+	response, err := r.ClustersMgmt().V1().Clusters().List().
+		Search(query).
+		Page(1).
+		Size(1).
+		SendContext(ctx)
+
+	if response.Total() == 1 {
+		return response.Items().Slice()[0], nil
+	}
+	return nil, fmt.Errorf("cluster %q not found: %v", clusterName, err)
+}
+
+// deleteCluster handles sending the request to delete the cluster
+func (r *Provider) deleteCluster(ctx context.Context, clusterID string) error {
+	if clusterID == "" {
+		return errors.New("cluster ID is undefined and is required")
+	}
+
+	r.log.Info("Initiating cluster deletion for", clusterIDLoggerKey, clusterID)
+
+	commandArgs := []string{"delete", "cluster", "--cluster", clusterID, "--yes"}
+
+	_, stderr, err := r.RunCommand(ctx, exec.CommandContext(ctx, r.rosaBinary, commandArgs...))
+	if err != nil {
+		return fmt.Errorf("error: %v, stderr: %v", err, stderr)
+	}
+
+	r.log.Info("Cluster", clusterIDLoggerKey, clusterID, "deletion initiated!")
+
+	return err
+}
+
+// waitForClusterToBeInstalled waits for the cluster to be in a ready state
+func (r *Provider) waitForClusterToBeInstalled(ctx context.Context, clusterID string, timeout time.Duration) error {
+	getClusterState := func() (string, error) {
+		commandArgs := []string{"describe", "cluster", "--cluster", clusterID, "--output", "json"}
+
+		stdout, stderr, err := r.RunCommand(ctx, exec.CommandContext(ctx, r.rosaBinary, commandArgs...))
+		if err != nil {
+			return "", fmt.Errorf("error: %v, stderr: %v", err, stderr)
+		}
+
+		output, err := cmd.ConvertOutputToMap(stdout)
+		if err != nil {
+			return "", fmt.Errorf("failed to convert output to map: %v", err)
+		}
+
+		clusterState := fmt.Sprint(output["status"].(map[string]any)["state"])
+
+		return clusterState, err
+	}
+
+	err := wait.For(func() (bool, error) {
+		clusterState, err := getClusterState()
+		if err != nil {
+			return false, err
+		}
+
+		if clusterState != "ready" {
+			r.log.Info("Cluster", clusterIDLoggerKey, clusterID, "not in ready state", clusterStateLoggerKey, clusterState)
+			return false, nil
+		}
+
+		r.log.Info("Cluster is ready!", clusterIDLoggerKey, clusterID)
+		return true, nil
+	}, wait.WithTimeout(timeout), wait.WithInterval(1*time.Minute))
+	if err != nil {
+		return fmt.Errorf("cluster %q failed to enter ready state in the alloted time", clusterID)
+	}
+	return nil
+}
+
+// waitForClusterToBeHealthy waits for the cluster health check job to succeed
+func (r *Provider) waitForClusterToBeHealthy(ctx context.Context, kubeconfigFile, reportDir string, hostedCP bool) error {
+	client, err := openshiftclient.NewFromKubeconfig(kubeconfigFile, r.log)
+	if err != nil {
+		return fmt.Errorf("failed to construct openshift client: %v", err)
+	}
```

Comment:
can we take a client instead of the kubeconfigfile? Seems to specific to the implementation

Reply:
Great suggestion, resolved in e7becc4e30f7106fb4fb72a56f2b068bcfe04f45